src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrameManager.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    using ConcurrentFramesDictionary = ConcurrentDictionary<MethodSymbol, MethodGroupConversionCacheFrame>;
+
+    internal sealed class MethodGroupConversionCacheFrameManager : CommonMethodGroupConversionCacheFrameManager, IComparer<MethodGroupConversionCacheFrame>
+    {
+        internal CSharpCompilation Compilation { get; }
+
+        private ConcurrentFramesDictionary _lazyFrames;
```

Comment:
I'd avoid the name Frames for this purpose. "Frame" is usually used for something related to stack frame. I'd call these "containers".